File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    def dfs(idx, cur):
        # 종료 조건
        if idx == len(numbers):
            if target == cur:
                return 1
            else:
                return 0
        
        cnt = 0
        nxt = numbers[idx]
        
        # +
        cnt += dfs(idx+1, cur + nxt)
        
        # -
        cnt += dfs(idx+1, cur - nxt)
        return cnt
    
    return dfs(0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int dfs(vector<int>& numbers, int& target, int idx, int current_sum) {
    if (idx == numbers.size()) {
        if (current_sum == target) return 1;
        else return 0;
    }
    
    int total = 0;
    total += dfs(numbers, target, idx + 1, current_sum + numbers[idx]);
    total += dfs(numbers, target, idx + 1, current_sum - numbers[idx]);
    return total;
}

int solution(vector<int> numbers, int target) {
    return dfs(numbers, target, 0, 0);
}